controllers/gcpcluster.go

```diff
@@ -70,6 +71,10 @@ func (r *GCPClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		log.Info("GCP Cluster does not have an owner cluster yet")
 		return ctrl.Result{}, nil
 	}
+	if *gcpCluster.Status.Network.SelfLink == "" {
```

Comment:
could `gcpCluster.Status.Network` be `nil` at this point? I'm worried it could panic due to a null pointer

Reply:
Unknow, the problem i have seen is that this value was empty, and if is empty and the empty string is passed then GCP is so smart it will not error but put the firewall rule into default VPC

Reply:
Would adding a check if it's not nil would be enough?
```
if  gcpCluster.Status.Network == nil || *gcpCluster.Status.Network.SelfLink == "" {
```

Reply:
Yeah , checking if it's `nil` would be enough I think